test/singlethreaded.jl

```diff
@@ -1,120 +1,121 @@
-using FactCheck
 using ReadWriteLocks
+using Test
 
-facts("Single-threaded tests") do
-    context("Initialization") do
+@testset "Single-threaded tests" begin
+    @testset "Initialization" begin
         rwlock = ReadWriteLock()
 
-        @fact rwlock.read_lock.rwlock --> rwlock
-        @fact rwlock.write_lock.rwlock --> rwlock
-        @fact rwlock.readers --> 0
-        @fact rwlock.writer --> false
-        @fact read_lock(rwlock) --> rwlock.read_lock
-        @fact write_lock(rwlock) --> rwlock.write_lock
+        @test rwlock.read_lock.rwlock == rwlock
+        @test rwlock.write_lock.rwlock == rwlock
+        @test rwlock.readers == 0
+        @test rwlock.writer == false
+        @test read_lock(rwlock) == rwlock.read_lock
+        @test write_lock(rwlock) == rwlock.write_lock
     end
 end
 
-facts("Two-threaded tests") do
-    context("Read locks") do
+@testset "Two-threaded tests" begin
+    @testset "Read locks" begin
         NUM_LOCKS = 10
 
-        context("$NUM_LOCKS locks") do
+        @testset "$NUM_LOCKS locks" begin
             rwlock = ReadWriteLock()
             rlock = read_lock(rwlock)
 
             @async begin
-                @fact rwlock.writer --> false
-                @fact rwlock.readers --> 0
+                @test rwlock.writer == false
+                @test rwlock.readers == 0
                 for i = 1:NUM_LOCKS
-                    @fact lock!(rlock) --> nothing
-                    @fact rwlock.readers --> i
+                    @test lock!(rlock) == nothing
```

Comment:
We don't need to do any tests of the return values of `lock` and `unlock` anymore. We can just call the functions.

FactCheck was not as good as Test at handling errors outside of tests.

Reply:
hmm, changing all cases of `@test [un]lock!(x) == nothing` -> `[un]lock!(x)` causes tests to fail -- we hit code we were not hitting before 🤔

Reply:
Push a commit that does that so I can see

Reply:
https://github.com/invenia/ReadWriteLocks.jl/pull/4/commits/b006f13c1c58bee2dbc0c5bd5734367a33f88bc6

Reply:
lol okay put them back I guess? Bizarre.

Reply:
reverted 🤷‍♂